videosink/encoder.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package videosink
+
+import (
+	"image/jpeg"
+	"image/png"
+	"sync"
+)
+
+var jpegOptions = jpeg.Options{
+	Quality: 95,
+}
+
+type pngEncoderBufferPool sync.Pool
+
+var _ png.EncoderBufferPool = (*pngEncoderBufferPool)(nil)
+
+func (p *pngEncoderBufferPool) Get() *png.EncoderBuffer {
+	if buf := (*sync.Pool)(p).Get(); buf != nil {
+		return buf.(*png.EncoderBuffer)
+	}
+
+	return nil
+}
+
+func (p *pngEncoderBufferPool) Put(buf *png.EncoderBuffer) {
+	(*sync.Pool)(p).Put(buf)
+}
+
+type pngEncoderManager struct {
+	once sync.Once
+	pool pngEncoderBufferPool
+	enc  *png.Encoder
+}
+
+var pngEncoder pngEncoderManager
+
+// get returns a PNG encoder with a globally shared buffer pool.
+func (m *pngEncoderManager) get() *png.Encoder {
+	m.once.Do(func() {
+		m.enc = &png.Encoder{
+			CompressionLevel: png.BestSpeed,
```

Comment:
I would expect this to be configurable in Options eventually, I can imagine case where folks may want to have maximum compression.

Reply:
Added a TODO to that effect in the Options structure. I'll do it in a follow-up change, just like keep-alive and rate-limiting.